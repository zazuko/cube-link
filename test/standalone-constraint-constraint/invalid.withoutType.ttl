@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix cube: <https://cube.link/> .
@prefix observation: <https://environment.ld.admin.ch/foen/nfi/observation/max_min_undefined> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix schema: <http://schema.org/> .
@prefix qudt: <http://qudt.org/schema/qudt/>  .
@base <https://example.org/> .

<cube> a cube:Cube ;
    cube:observationConstraint <shape> ;
    cube:observationSet <observationSet> .

<observationSet> cube:observation <observationA> , <observationB> .

<observationA> a cube:Observation ; cube:observedBy <observer> ;
    <dimension1> 4.9 .

<observationB> a cube:Observation ; cube:observedBy <observer> ;
    <dimension1> 1.0 .

<shape> a cube:Constraint ;
    sh:targetClass cube:Observation ;
    sh:closed true ;
    sh:property [ sh:path rdf:type ; sh:nodeKind sh:IRI] ;
    sh:property [ sh:path cube:observedBy ; sh:nodeKind sh:IRI ] ;
    sh:property [
        sh:path  <dimension1> ;
        #sh:datatype xsd:decimal ;
        schema:name "some Quantity" ;
    ] ;
.