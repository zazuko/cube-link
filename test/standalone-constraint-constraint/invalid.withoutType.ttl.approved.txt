@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix cube: <https://cube.link/> .

_:report a sh:ValidationReport ;
	sh:result [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Violation ;
		sh:sourceConstraintComponent sh:NodeConstraintComponent ;
		sh:sourceShape [
			sh:path sh:property ;
			sh:node <https://cube.link/shape/ObservationConstraintProperty> ;
			sh:message "The constraints do not validate" ;
		] ;
		sh:focusNode <https://example.org/shape> ;
		sh:value _:b3 ;
		sh:resultMessage "The constraints do not validate" ;
		sh:detail [
			rdf:type sh:ValidationResult ;
			sh:resultSeverity sh:Violation ;
			sh:sourceConstraintComponent sh:OrConstraintComponent ;
			sh:sourceShape [
				sh:message "needs a sh:datatype, sh:nodeKind or sh:datatype within sh:or (...)" ;
				sh:or (
					[
						sh:path sh:datatype ;
						sh:minCount 1 ;
					]
					[
						sh:path sh:nodeKind ;
						sh:minCount 1 ;
					]
					[
						sh:path sh:or ;
						sh:node <https://cube.link/shape/listnode>, [
							sh:property [
								sh:path sh:datatype ;
								sh:minCount 1 ;
							] ;
							sh:path (
								[
									sh:zeroOrMorePath rdf:rest ;
								]
								rdf:first
							) ;
						] ;
						sh:minCount 1 ;
					]
				) ;
			] ;
			sh:focusNode _:b3 ;
			sh:value _:b3 ;
			sh:resultMessage "needs a sh:datatype, sh:nodeKind or sh:datatype within sh:or (...)" ;
		] ;
		sh:resultPath sh:property ;
	] ;
	sh:conforms false .
